### ISSUE TITLE
Is it possible to move the labels closer to the nodes in Flowcharts?

### ISSUE THREAD

**Issue by MechMK1** [mermaid-js/mermaid#1233]
I would like to create a flowchart with text on a connecting line. However, sometimes the lines are very long and the text of the line is in the middle of the line, rather than close to the node with the decision. Here is an example to illustrate this:

![Image of Yaktocat](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgICBEZWNpc2lvbntTb21lIERlY2lzaW9ufVxuICAgIENvbW1vbihOZXh0IENvbW1vbiBUaGluZylcblxuICAgIERlY2lzaW9uIC0tWWVzLS0-IEEgLS0-IEIgLS0-IEMgLS0-IEQgLS0-IENvbW1vblxuICAgIERlY2lzaW9uIC0tTm8tLT4gQ29tbW9uXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCIsImZsb3djaGFydCI6eyJjdXJ2ZSI6ImxpbmVhciJ9fSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)

The `No` text is in the middle of the line, rather than being close to the "Decision" node. Is there a way to get the text closer to the "Decision" node, similar to how `Yes` is positioned?

**Comment by Philzen** [comment]
This would be a great addition – Mermaid is awesome in its declarativeness and ease of use, but this really was a bump for me.

In more complex flowcharts, it makes them simply harder to process visually, because you want the decision label next to the decision, not somewhere down along the line:

```mermaid
flowchart-elk LR
    A(("Gitlab runner\nstarting\n (Java + Maven\nproject)")) --> B
    
    B{<code>.sdkmanrc</code> ?}
    B --Yes----> LocalMatch
    B -->|No| X

    LocalMatch{Required Java &<br>Maven versions<br>present?} 
    LocalMatch -->|Yes| X
    LocalMatch -->|No| Sdkman
    
    Sdkman{sdkman\npresent?} 
    Sdkman -->|Yes| G
    Sdkman -->|No| Install

    Install[Install\nsdkman] --> G

    G[Run <code>sdk env</code>] --> X

    X((Execute Job))

```

For example see the `No` label for the first decision or the `Yes` label for the second one. Your eyes need to wander down the paths to see why you would take them – while you want to know what path leads where, right away, next to the decision itself.

If this is considered a bug, well than the expectation and fix is that labels connected to the outputs of a rhombus node (`{decision}`) always have to be rendered next to it.

If this is considered a feature request and would be fixed by introducing another declarative feature, i feel this could be tackled (or at least should be harmonized with) https://github.com/mermaid-js/mermaid/issues/1652

**Comment by megantriplett** [comment]
Hi, I would like to work on this issue with @udvale

**Comment by sidharthv96** [comment]
Just to be clear, there's no need to wait for issues to be assigned before working on them. If you're interested in working on an issue, you can leave a comment so that others can see it. PRs are the way we usually link issues to contributors. So you can directly open a PR with the issue number in title/description, and GitHub will link those. 

The reason why we don't assign issues is that it adds a lot of friction, waiting for someone to assign issues, work gets abandoned (life happens, and we're all doing it for free, so no issues with it), and others won't pick up the issue if it's already assigned. It's hard for maintainers to keep track of abandoned issues and unassign them, etc. 

TLDR; There's no need to request issues to be assigned to you, just raise the PR directly!

**Comment by megantriplett** [comment]
Would this:

![Image](https://github.com/user-attachments/assets/d003dec0-672b-4c3c-a8d0-918cb8786bb2)

be an acceptable movement? Or should we (@udvale) continue to tinker our code so that the edges are moved differently?

**Comment by udvale** [comment]
Hi @sidharthv96, me and @megantriplett are facing an issue with edge label placement when dealing with both vertical (TD) and horizontal (LR) flowcharts. Specifically:

For vertical (TD) graphs, adjusting the y offset works fine to correctly position edge labels (shown in above graph).
However, the same adjustment affects horizontal (LR) graphs, causing labels to be placed out of scope or incorrectly aligned.
To fix this, I believe we need to identify the flowchart direction (TD or LR) from the chart definition (e.g., flowchart TD or flowchart LR) and apply direction-specific adjustments.

My main difficulty is figuring out how to retrieve the diagram's direction programmatically within the codebase. Currently, I'm considering extracting it from the chart definition during parsing or configuration.
```HTML
flowchart TD /* or this can be also flowchart LR */
        A[Christmas] -->|Get money| B(Go shopping)
        A --> |Vroom| F[fa:fa-car Car]
        B --> C{Let me think}
        C -->|One| D[Laptop]
        C -->|Two| E[iPhone]
        C -->|Three| F[fa:fa-car Car]
```

Am I on the right track with this approach, or is there a better way to handle this? Any guidance on where to look in the codebase to access the diagram direction would be greatly appreciated! 😊

**Comment by sidharthv96** [comment]
The layout looks good to me. 
@udvale the direction would be available in the flowDB's `getDirection` function.

**Comment by megantriplett** [comment]
![Image](https://github.com/user-attachments/assets/6caf02d4-fc05-4636-8be1-2424c07a0eea)


@sidharthv96, We've (@udvale and I) tried the above three ways to access the direction, with no success. Is there an object in edges.js that represents the flowchart? We've been unable to identify which object we should call the getDirection function on.

**Comment by sidharthv96** [comment]
Hi @megantriplett, which file are you editing?

This is the function I was referring to. 

https://github.com/mermaid-js/mermaid/blob/ad2f17205aecb0730e1472f35142931086026411/packages/mermaid/src/diagrams/flowchart/flowDb.ts#L544-L546

**Comment by megantriplett** [comment]
Hi @sidharthv96, We're making edits in the mermaid\packages\mermaid\src\rendering-util\rendering-elements\edges.js file

**Comment by jrgilman** [comment]
Did this go anywhere? It's pretty frustrating and heavily effects our use of the online platform...
